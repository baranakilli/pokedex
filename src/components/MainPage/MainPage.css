@import url('https://fonts.googleapis.com/css2?family=Bungee+Inline&display=swap');

h1{
  font-family: 'Bungee Inline', cursive;
  color: white;
  font-weight: 400;
  font-size: min(10vw, 80px);
  margin: 15px 0 15px 0;
}
