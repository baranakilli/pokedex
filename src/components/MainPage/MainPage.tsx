import { ChangeEvent, Component } from 'react';
import Header from '../Header';
import SearchBox from '../SearchBox';
import CounterButton from '../CounterButton/CounterButton';
import Scroll from '../Scroll';
import ErrorBoundry from '../ErrorBoundry';
import CardList from '../CardList/CardList';
import './MainPage.css';

import { Robot } from '../../actions';

interface MainPageProps {
  onRequestRobots: () => void;
  onSearchChange: (event: ChangeEvent<HTMLInputElement>) => void;
  robots: Robot[];
  searchField: string;
  isPending: boolean;
  error?: string | Error;
}

class MainPage extends Component<MainPageProps> {
  
  componentDidMount() {
    this.props.onRequestRobots();
  }
  filterRobots = () => {
    const { robots, searchField } = this.props;
    return robots.filter((robot) => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
    });
  };

  render() {
    const { onSearchChange, isPending, error } = this.props;

    return (
      <div className="tc">
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <CounterButton />
        <Scroll>
          <ErrorBoundry>
            {isPending ? (
              <h1 className="tc f1">LOADING..</h1>
            ) : error ? (
              <h1 className="tc dark-red">ERROR</h1>
            ) : (
              <CardList robots={this.filterRobots()} />
            )}
          </ErrorBoundry>
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
