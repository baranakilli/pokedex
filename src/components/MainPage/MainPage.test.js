import { shallow } from "enzyme";
import MainPage from "./MainPage";

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRequestPokemons: jest.fn(),
    pokemons: [],
    searchField: "",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

it("renders without crashing", () => {
  expect(wrapper).toMatchSnapshot();
});

it("filters Pokemons", () => {
  const mockProps = {
    onRequestPokemons: jest.fn(),
    pokemons: [],
    searchField: "a",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper.instance().filterPokemons()).toEqual([]);
});

it("filters Pokemons correctly", () => {
  const filteredPokemons = [
    {
      id: 1,
      name: "Leanne Graham",
      username: "Bret",
      email: "Sincere@april.biz",
    },
  ];
  const mockProps = {
    onRequestPokemons: jest.fn(),
    pokemons: [
      {
        id: 1,
        name: "Leanne Graham",
        username: "Bret",
        email: "Sincere@april.biz",
      },
    ],
    searchField: "Leanne",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper.instance().filterPokemons()).toEqual(filteredPokemons);
});

it("filters Pokemons correctly 2", () => {
  const mockProps = {
    onRequestPokemons: jest.fn(),
    pokemons: [
      {
        id: 1,
        name: "Leanne Graham",
        username: "Bret",
        email: "Sincere@april.biz",
      },
    ],
    searchField: "Xavier",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper.instance().filterPokemons()).toEqual([]);
});

it("isPending working", () => {
  const mockProps = {
    onRequestPokemons: jest.fn(),
    pokemons: [],
    searchField: "a",
    isPending: true,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper).toMatchSnapshot();
});

it("error display working", () => {
  const mockProps = {
    onRequestPokemons: jest.fn(),
    pokemons: [],
    searchField: "a",
    isPending: false,
    error: true,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper).toMatchSnapshot();
});
