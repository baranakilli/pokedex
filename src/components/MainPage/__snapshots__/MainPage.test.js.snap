// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error display working 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <h1
        className="tc dark-red"
      >
        ERROR
      </h1>
    </ErrorBoundry>
  </Scroll>
</div>
`;

exports[`isPending working 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <h1
        className="tc white f1"
      >
        Loading..
      </h1>
    </ErrorBoundry>
  </Scroll>
</div>
`;

exports[`renders without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        pokemons={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
